chainer/functions/loss/sigmoid_cross_entropy.py

```diff
@@ -23,48 +26,44 @@ def check_type_forward(self, in_types):
             x_type.shape == t_type.shape
         )
 
-    def forward_cpu(self, inputs):
+    def forward(self, inputs):
+        xp = cuda.get_array_module(*inputs)
         x, t = inputs
-        self.y, = sigmoid.Sigmoid().forward_cpu((x,))
+        self.ignore_mask = (t != self.ignore_label)
+        if self.normalize:
+            count = int(self.ignore_mask.sum())
+        else:
+            count = x.shape[0]
+        self.count = count
         # stable computation of the cross entropy.
-        loss = -numpy.sum(
-            x * (t - (x >= 0)) - numpy.log1p(numpy.exp(-numpy.abs(x))))
-        return numpy.array(loss / t.shape[0], dtype=x.dtype),
+        loss = -xp.sum(
+            self.ignore_mask * (x * (t - (x >= 0)) -
+                                xp.log1p(xp.exp(-xp.abs(x)))))
+        return xp.array(loss / self.count, dtype=x.dtype),
```

Comment:
`self.count` is 0 when all training data is ignored, which causes 0 division error. We must handle such a case.